### PR TITLE
STAR-84 Show User's Own Questions In Their Profile Page

### PR DESCRIPTION
**Pull request by farzaneh-haghani**
- Create a Page :
- ```Question``` 
- Created /api/questions/user/:userId   Backend Endpoint
- Created /api/questions/:questionId   Backend Endpoint

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -51,6 +52,14 @@ const router = createBrowserRouter(
           </Protected>
         }
       />
+      <Route
+        path="questions/:id"
```
Comment (bazmurphy) [inline client/src/App.jsx:56]:
```suggestion
        path="questions/:questionId"
```

File: client/src/pages/Question.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+import { NavLink, useParams } from "react-router-dom";
```
Comment (bazmurphy) [inline client/src/pages/Question.jsx:2]:
```suggestion
import { useParams } from "react-router-dom";
```

File: client/src/pages/Question.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { Box, Button, Typography } from "@mui/material";
```
Comment (bazmurphy) [inline client/src/pages/Question.jsx:3]:
```suggestion
import { Box, Typography } from "@mui/material";
```

File: client/src/pages/Question.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+const Question = () => {
+  const [question, setQuestion] = useState("");
+  const id = useParams().id;
```
Comment (bazmurphy) [inline client/src/pages/Question.jsx:7]:
```suggestion
  const questionId = useParams().questionId;
```

File: client/src/pages/Question.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  useEffect(() => {
+    const openHandler = async (id) => {
```
Comment (bazmurphy) [inline client/src/pages/Question.jsx:10]:
```suggestion
    const fetchQuestionById = async (questionId) => {
```

File: client/src/pages/Question.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      try {
+        const response = await fetch(
+          `${import.meta.env.VITE_SERVER_URL}/api/questions/${id}`,
```
Comment (bazmurphy) [inline client/src/pages/Question.jsx:13]:
```suggestion
          `${import.meta.env.VITE_SERVER_URL}/api/questions/${questionId}`,
```

File: client/src/pages/Question.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+          }
+        );
+        const data = await response.json();
+        // console.log(data);
+        setQuestion(data);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    openHandler(id);
```
Comment (bazmurphy) [inline client/src/pages/Question.jsx:25]:
```suggestion
    fetchQuestionById(questionId);
```

File: client/src/pages/Profile.jsx [hunk 8]
```diff
@@ -1,17 +1,50 @@
...
+import { useContext, useState, useEffect } from "react";
+import { Box, Card, CardMedia, ListItem, Typography } from "@mui/material";
 import { AuthContext } from "../context/AuthContext";
+import { NavLink } from "react-router-dom";
 
 const Profile = () => {
   // get the userCookie from AuthContext
   const { userCookie } = useContext(AuthContext);
 
+  // define state to store the Users Data
```
Comment (bazmurphy) [inline client/src/pages/Profile.jsx:10]:
```suggestion
  // define state to store the Users Question Data
```

File: client/src/pages/Profile.jsx [hunk 9]
```diff
@@ -1,17 +1,50 @@
...
+  const [userQuestionsData, setUserQuestionsData] = useState(null);
+
+  useEffect(() => {
+    // fetch the Users Data from the backend
```
Comment (bazmurphy) [inline client/src/pages/Profile.jsx:14]:
```suggestion
    // fetch the Users Question Data from the backend
```

File: server/src/controllers/questionsController.ts [hunk 10]
```diff
@@ -47,3 +47,45 @@ export const addQuestion = async (req: Request, res: Response) => {
     res.status(500).json(error);
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:47]:
```suggestion
    res.status(500).json({ error: "Server Error" })
```

please while we are here anyway 😁

**Review by bazmurphy — COMMENTED** [review]
Gooooood! 😎⭐

Works welll

A few tiny changes for clarity and that's all.

File: client/src/App.jsx (hunk 1 above)
Comment (jarrodruby) [reply client/src/App.jsx:56]: The convention used by most frameworks is `id`. Look at the [Rails Routing](https://guides.rubyonrails.org/routing.html) doc for example. Idea is that it correlates to the primary key

**Review by bazmurphy — APPROVED** [review]
Great Job 😎 👍 ⭐

**Pull request merged**